examples/vision/metric_learning.py

```diff
@@ -1,18 +1,20 @@
 """
 Title: Metric learning for image similarity search
-Author: [Mat Kelcey](https://twitter.com/mat_kelcey)
-Date created: 2020/06/05
-Last modified: 2020/06/09
+Author: [Owen Vallis](https://twitter.com/owenvallis)
```

Comment:
This appears to be a very extensive rewrite of the example. Prefer creating a new example besides the old one rather than overwriting the old one. Only if the changes are relatively limited should we update an existing example.

Reply:
Sounds good, I'll rename it and add the original example back. Given that the two examples are very similar, should I name this one so that it's clear they are related, maybe something like `metric_learning_tf_similarity.py`?

Reply:
Sounds good! Definitely do point out "TF Similarity" in the title as the differentiating aspect of this example.

Reply:
Done